pkg/apis/infinispan/v1/infinispan_types.go

```diff
@@ -47,16 +47,16 @@ const (
 
 // InfinispanServiceSpec specify configuration for specific service
 type InfinispanServiceSpec struct {
-	Type      ServiceType                    `json:"type"`
-	Container InfinispanServiceContainerSpec `json:"container"`
-	Sites     InfinispanSitesSpec            `json:"sites"`
+	Type      ServiceType                     `json:"type"`
+	Container *InfinispanServiceContainerSpec `json:"container,optional,omitempty"`
+	Sites     *InfinispanSitesSpec            `json:"sites,optional,omitempty"`
 }
 
 // InfinispanContainerSpec specify resource requirements per container
 type InfinispanContainerSpec struct {
-	ExtraJvmOpts string `json:"extraJvmOpts"`
-	Memory       string `json:"memory"`
-	CPU          string `json:"cpu"`
+	ExtraJvmOpts string `json:"extraJvmOpts,omitempty"`
+	Memory       string `json:"memory,omitempty"`
```

Comment:
I don't agree with CPU and Memory being `omitempty` as they are always set to some value. Either user provided or default but never empty. It'd be good to propagate this information back to user.